File: programmers/level_1/77484.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

vector<int> solution(vector<int> lottos, vector<int> win_nums) {
    vector<int> answer;
    set<int> lotto;
    for (int i = 0; i < lottos.size(); i++) {
        if (lottos[i] != 0) {
            lotto.insert(lottos[i]);
        }
    }

    set<int> win_num;
    for (int i = 0; i < win_nums.size(); i++) {
        win_num.insert(win_nums[i]);
    }

    int min = 0;
    int another = 0;
    for (auto i: lotto) {
        if (win_num.find(i) != win_num.end()) {
            min++;
        } else {
            another++;
        }
    }

    answer.push_back(another == 6 ? 6 : 7 - (6 - another));
    answer.push_back(min == 0 ? 6 : 7 - min);

    return answer;
}

File: kotlin_ps/src/main/kotlin/programmers/programmers_77484/77484.kt
package programmers.programmers_77484

class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        var answer = IntArray(2) { 0 }

        var correctCount = 0
        var zeroCount = 0
        for (lotto in lottos) {
            if (lotto == 0) {
                zeroCount++
                continue
            }
            if (win_nums.contains(lotto)) {
                correctCount++
            }
        }
        answer[0] = if (correctCount + zeroCount == 0) {
            6
        } else {
            7 - (correctCount + zeroCount)
        }
        answer[1] = if (correctCount == 0) {
            6
        } else {
            7 - correctCount
        }

        return answer
    }
}